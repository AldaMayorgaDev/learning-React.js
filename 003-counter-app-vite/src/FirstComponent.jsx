import PropTypes from 'prop-types';

const getSaludo = ()=>{
  return 'Hola Aldahir';
}
export const FirstComponent = ({title, mensaje= 'Legend of zelda Tears of the Kingdom',edad  }) => {
  
  if (!title) {
    throw new Error('el title no existe.')
  }
  
  console.log('props.title', title)
  return (
    <>
      <h2>{title}</h2>
      <h3>{mensaje}</h3>
      <p>{getSaludo()}</p>
      <p>{edad + 10}</p>
    </>
  )
};

FirstComponent.propTypes = {
    title: PropTypes.string.isRequired,
    mensaje: PropTypes.string,
    edad : PropTypes.number.isRequired
}

FirstComponent.defaultProps = {
  title: 'No hay titulo',
  edad: 0,
  mensaje: 'The Legend of Zelda: Tears of the kingdom'
}