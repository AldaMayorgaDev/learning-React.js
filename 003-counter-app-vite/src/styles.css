html, body {
    background-color: #21232A;
    color: white;
    font-family: Helvetica, arial, sans-serif;
    font-size: 16px;
    padding: 70px;
}

button {
    padding:15px;
    font-size: 1.3rem;
    border-radius: .5rem;
    margin-right: 15px;
}