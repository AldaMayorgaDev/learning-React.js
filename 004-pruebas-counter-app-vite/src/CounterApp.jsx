//importando hooks
import { useState } from 'react';

//Importando Fragment solo pro practica puede remplazarse por el uso de <></>
// Importando PropTypes para definir tipo de datos de las propiedades
import { Fragment } from "react";
import PropTypes from "prop-types";


//functional componente CounterApp que recibe value

export const CounterApp =({value})=>{

    const [ counter , setCounter ] = useState ( value );


    const handleAdd = ()=>{ 
    
        setCounter( counter +1 );
    }

    const handleSubtract = ()=>{
        setCounter( counter - 1);
    }

    const handleReset = ()=>{
        setCounter( value );
    }
    return (
        <Fragment>
            <h1>CouterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={  ( )=>{ handleAdd()}  }>
                +1
            </button>

            <button onClick={handleSubtract}>
                 -1 
            </button>


            <button aria-label="btn-reset" onClick={handleReset}> 
                Reset 
            </button>
        </Fragment>
    );
}

// Definicion de propTypes para value
CounterApp.propTypes ={
    value: PropTypes.number.isRequired
}

// Definicion de default props
CounterApp.defaultProps = {
    value: 0,
}