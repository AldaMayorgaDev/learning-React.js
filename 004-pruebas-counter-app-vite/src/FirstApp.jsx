
//Exportando componente
export function App() {
    return <h1>Hola Mundo:!</h1>;
}


//Exportando componente Ocupando el shorthoot rafc
export const FirstApp = () => {
  return (
    <h1>Hello word</h1>
  )
}
