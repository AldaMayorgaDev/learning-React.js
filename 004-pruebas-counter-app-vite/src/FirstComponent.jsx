import PropTypes from 'prop-types';


export const FirstComponent = ({title, subtitle,name  }) => {
  
  return (
    <>
      <h1 data-testid="test-title">{title}</h1> {/* Se a√±ade el atibuto data-testid para buscar en las pruebas con getByTestId() */}
      <p>{subtitle}</p>
      <p>{subtitle}</p>
      <p>{name }</p>
    </>
  )
};

FirstComponent.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    name : PropTypes.string.isRequired
}

FirstComponent.defaultProps = {
  name: 'Aldahir',
  subtitle: 'No hay subtitulo',
}