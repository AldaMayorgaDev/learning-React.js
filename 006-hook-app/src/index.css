body{
    padding: 20px;
}

button{
    margin-right: 10px;
}

/* a{
    color: #131313;
    text-decoration: none;
}

a:active{
    color: #69696A;
    text-decoration: none;
} */

/* .navbar-brand, .nav-link{
    padding-left: 10px
} */