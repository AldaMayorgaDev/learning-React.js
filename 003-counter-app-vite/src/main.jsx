import React from 'react';
import ReactDOM from 'react-dom/client';
//import {App, FirstApp} from './FirstApp';
//import { FirstComponent } from './FirstComponent';
import { CounterApp } from './CounterApp';

//Importar estilos
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp  value={1}/>
    </React.StrictMode>
);