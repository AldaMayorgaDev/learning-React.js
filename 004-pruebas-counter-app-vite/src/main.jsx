import React from 'react';
import ReactDOM from 'react-dom/client';
//import {App, FirstApp} from './FirstApp';
//import { FirstComponent } from './FirstComponent';
import { CounterApp } from './CounterApp';

//Importar estilos
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp  value={100}/>
       {/*  <FirstComponent title="Legend of zelda: tears of the kingdom" /> */}
    </React.StrictMode>
);