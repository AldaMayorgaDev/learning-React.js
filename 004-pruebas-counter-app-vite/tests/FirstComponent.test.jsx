import { render } from '@testing-library/react';
import { FirstComponent } from '../src/FirstComponent';

describe('Pruebas en <FirstComponent />', () => { 

    test('Debe de hacer match con el snapshot', () => { 

        //1.- llamar funcion render de la dependenciade react testing library, importandola
        // render es una funcion que literamente renderiza el componente en memoria

        const title = 'Legend of Zelda: Breath of the Wild';
        const {container }= render( <FirstComponent title={title}/>);
       

        /* toMatchsnapshot crear la carpeta __snapshots__ y en ella un archivo con el html que se obtiene y queda ahi para compararlo y validar la prueba */
         expect(container).toMatchSnapshot();
     })


     test('Debe de mostrar el title en un h1', () => { 
        const title = 'Legend of Zelda: Breath of the Wild';
        const {container,  getByText, getByTestId}= render( <FirstComponent title={title}/>);


        expect(getByText(title)).toBeTruthy(); //Valida que si exista el texto del title en el dom renderizado


        //Esta forma no se recomienda pero es posible en ciertos casos

     /*    const h1 = container.querySelector('h1'); //selecionamos la etiqueda h1 del dom
        expect(h1.innerHTML).toBe(title) //se espera que el contenido de la etiqued h1 por eso el innerHTML, sea igual a title */



        //Esta forma si re recoomienda
        /* 1.- Agregar en el componente el atributo data-testid="xxx" 
           2.- extrar del objeto de render(), getByTestId
           3.- Hacer la asercion utilizando getByTestId('xxx')
        */

           expect(getByTestId('test-title').innerHTML).toBe(title);

      })


      test('Debe de mostrar el sutitle enviado por props', () => { 

        const title = 'Legend of Zelda: Breath of the Wild';
        const subtitle = 'Soy un subtitulo de testing';

        const {getByText}= render(
            <FirstComponent 
                title={title} 
                subtitle={subtitle}
            />
        );

        expect(getByText(subtitle)).toBeTruthy();
            
      })
 })