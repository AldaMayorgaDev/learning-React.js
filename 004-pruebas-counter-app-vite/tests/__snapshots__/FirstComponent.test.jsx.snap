// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstComponent /> Debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Legend of Zelda: Breath of the Wild
  </h1>
   
  <p>
    No hay subtitulo
  </p>
  <p>
    Aldahir
  </p>
</div>
`;
